framework/encode/capture_settings.cpp

```diff
@@ -697,5 +708,20 @@ std::string CaptureSettings::ParseTrimKeyString(const std::string& value_string)
     return trim_key;
 }
 
+uint32_t CaptureSettings::ParseTrimKeyFramesString(const std::string& value_string)
+{
+    uint32_t frames = 0;
+    int      value  = std::stoi(value_string);
```

Comment:
`std::stoi` will throw an exception if the string contains something unparseable (`invalid_argument` I think).

You can test this in your branch with `GFXRECON_CAPTURE_TRIGGER_FRAMES=a1` or some such.

CaptureSettings::ParseTrimRangeString validates that frame numbers contain only digits; would you please add code similar to that to your new function `ParseTrimKeyFramesString`?

(`GFXRECON_CAPTURE_TRIGGER_FRAMES=1a` doesn't throw, but that probably should also print an error.)

Reply:
Thanks!
I had missed that `ParseTrimKeyFramesString` was validating the string was digits.